includes/class-amp-theme-support.php

```diff
@@ -1834,7 +1836,23 @@ public static function is_output_buffering() {
 	 */
 	public static function finish_output_buffering( $response ) {
 		self::$is_output_buffering = false;
-		$response                  = self::prepare_response( $response );
+
+		try {
+			$response = self::prepare_response( $response );
+		} catch ( Exception $exception ) {
+			$title   = __( 'Failed to prepare AMP response', 'amp' );
+			$message = __( 'A server error occurred while trying to prepare the AMP response.', 'amp' );
+
+			/** @var ErrorPage $error_page */
+			$error_page = Services::get( 'error_page' );
+
+			$error_page
```

Comment:
The output buffer needs to be overridden with the error page:

```suggestion
			$response = $error_page
```

Reply:
Hmm, there was a return there before. I seem to have dropped that at some point.

Reply:
Oh, no, I see now what happened.